### PR TITLE
ISSUE-11: Add ZIP checking/extraction

### PR DESCRIPTION
**Pull request by DiegoPino**
WIP still.

In case there is an attached ZIP and and every other attempt to fetch a local file fails we try with ZIP. 
Probably can be more strict and only do it if the `Path` is not absolute (Will do that tomorrow probably, may also sleep tonight)

Added some general DCS passes but may need more of that since these classes are a bit large and i'm tired and i do not trust the IDE... anyways

### CODE REVIEW

File: src/AmiUtilityService.php [hunk 1]
```diff
@@ -444,18 +489,110 @@ public function retrieve_remote_file(
...
+        if ($contents && file_put_contents($realpath, $contents)) {
+          return $realpath;
+        }
+      }
+      else {
+        // Opening the ZIP file failed.
+        return FALSE;
+      }
+    } catch (\Exception $exception) {
+      //@TODO what to do?
```
Comment (pcambra) [inline src/AmiUtilityService.php:565]: A watchdog_exception could be a good thing to throw in here

File: src/AmiUtilityService.php [hunk 2]
```diff
@@ -827,19 +975,25 @@ public function checkNodeAccess(string $bundle, AccountInterface $account = NULL
...
-   * @param $bundle
+   * @param                                            $bundle
    *
    * @return bool
    */
   public function checkFieldAccess($bundle, AccountInterface $account = NULL) {
     $fields = [];
     $access_handler = $this->entityTypeManager->getAccessControlHandler('node');
-    $field_definitions = $this->strawberryfieldUtility->getStrawberryfieldDefinitionsForBundle($bundle);
+    $field_definitions
```
Comment (pcambra) [inline src/AmiUtilityService.php:985]: Is this broken into lines intentionally?

File: src/AmiUtilityService.php [hunk 3]
```diff
@@ -641,10 +783,7 @@ public function csv_read(File $file) {
    * @return array
    * combined array
    */
-  public function ami_array_combine_special(
-    $header,
-    $row
-  ) {
+  public function ami_array_combine_special($header, $row) {
```
Comment (pcambra) [inline src/AmiUtilityService.php:786]: I think we should be using camelCase for these, but changing the signature of a public method probably not the best course of action

File: src/AmiUtilityService.php (hunk 1 above)
Comment (DiegoPino) [reply src/AmiUtilityService.php:565]: @pcambra I agree. I'm struggling with the fact we need to have a special type of logging in place that allows us a mix of persistence/context based displayable (means at an AMI set) for this things. My second pass now in the morning will be all about logging.

File: src/AmiUtilityService.php (hunk 2 above)
Comment (DiegoPino) [reply src/AmiUtilityService.php:985]: Probably not, IDE did some funky 80 lines max wrapping last night. On it

File: src/AmiUtilityService.php (hunk 3 above)
Comment (DiegoPino) [reply src/AmiUtilityService.php:786]: Will also do later, even if public nobody else is using this right now, so can move the functions (good thing of being an 0.1.0)

**Comment by DiegoPino** [comment]
Code review addressed and also added this tiny little thing here when a file is not found.

image

**Comment by DiegoPino** [comment]
My commit messages are not making any sense right now. But this is tested with a lot of variations and all is working. Links and warnings are better now. Merging. Thanks!

**Pull request merged**